File: mobile/src/services/payment.ts
import apiClient from './api';

export interface PaymentIntent {
  client_secret: string;
  payment_intent_id: string;
  amount: number;
  currency: string;
}

export interface PaymentRecord {
  id: string;
  booking_id: string;
  amount: number;
  currency: string;
  status: string;
  payment_method?: string;
  last_four?: string;
  card_brand?: string;
  created_at: string;
}

export interface PayoutSummary {
  total_earnings: number;
  pending_payouts: number;
  completed_payouts: number;
  this_month_earnings: number;
}

export const paymentService = {
  async createPaymentIntent(bookingId: string): Promise<PaymentIntent> {
    const response = await apiClient.post<PaymentIntent>('/payments/create-payment-intent', {
      booking_id: bookingId,
    });
    return response.data;
  },

  async confirmPayment(paymentIntentId: string): Promise<{ message: string; status: string }> {
    const response = await apiClient.post<{ message: string; status: string }>(
      '/payments/confirm-payment',
      null,
      { params: { payment_intent_id: paymentIntentId } }
    );
    return response.data;
  },

  async requestRefund(
    paymentId: string,
    amount?: number,
    reason?: string
  ): Promise<any> {
    const response = await apiClient.post('/payments/refund', {
      payment_id: paymentId,
      amount,
      reason,
    });
    return response.data;
  },

  async getMyPayments(): Promise<PaymentRecord[]> {
    const response = await apiClient.get<PaymentRecord[]>('/payments/my-payments');
    return response.data;
  },

  async getOwnerPayouts(): Promise<any[]> {
    const response = await apiClient.get<any[]>('/payments/owner/payouts');
    return response.data;
  },

  async getPayoutSummary(): Promise<PayoutSummary> {
    const response = await apiClient.get<PayoutSummary>('/payments/owner/summary');
    return response.data;
  },

  async createConnectAccount(country: string = 'US'): Promise<{ account_id: string; onboarding_url: string }> {
    const response = await apiClient.post<{ account_id: string; onboarding_url: string }>(
      '/payments/connect/create-account',
      { country, business_type: 'individual' }
    );
    return response.data;
  },
};

File: backend/app/schemas/payment.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel
from uuid import UUID

from app.models.payment import PaymentStatus, PayoutStatus

class PaymentIntentCreate(BaseModel):
    booking_id: UUID

class PaymentIntentResponse(BaseModel):
    client_secret: str
    payment_intent_id: str
    amount: int
    currency: str

class PaymentConfirm(BaseModel):
    payment_intent_id: str

class PaymentResponse(BaseModel):
    id: UUID
    booking_id: UUID
    amount: int
    currency: str
    status: PaymentStatus
    payment_method: Optional[str] = None
    last_four: Optional[str] = None
    card_brand: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

class RefundRequest(BaseModel):
    payment_id: UUID
    amount: Optional[int] = None  # Partial refund amount, None for full refund
    reason: Optional[str] = None

class RefundResponse(BaseModel):
    id: UUID
    original_amount: int
    refund_amount: int
    status: PaymentStatus
    refunded_at: datetime

class PayoutResponse(BaseModel):
    id: UUID
    owner_id: UUID
    booking_id: Optional[UUID] = None
    amount: int
    currency: str
    status: PayoutStatus
    processed_at: Optional[datetime] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

class PayoutSummary(BaseModel):
    total_earnings: int
    pending_payouts: int
    completed_payouts: int
    this_month_earnings: int

class ConnectAccountCreate(BaseModel):
    """For Stripe Connect onboarding."""
    country: str = "US"
    business_type: str = "individual"

class ConnectAccountResponse(BaseModel):
    account_id: str
    onboarding_url: str
